### PR TITLE
Context: remove network parameters

### PR DESCRIPTION
**Pull request by schildbach**
This is the first step in the goal of making Context optional again,
using it only for configuration of a test environment.

### CODE REVIEW

**Comment by msgilligan** [comment]
Concept Ack! Optional context would be a big improvement IMO.

**Comment by schildbach** [comment]
I kind of extracted two PRs from this PR: #2488 and #2489.

**Comment by schildbach** [comment]
Rebased.

File: core/src/test/java/org/bitcoinj/core/AbstractFullPrunedBlockChainTest.java [hunk 1]
```diff
@@ -234,13 +234,12 @@ public void testFinalizedBlocks() throws Exception {
...
     @Test
     public void testFirst100KBlocks() throws Exception {
-        Context context = new Context(MAINNET);
         File blockFile = new File(getClass().getResource("first-100k-blocks.dat").getFile());
         BlockFileLoader loader = new BlockFileLoader(MAINNET, Arrays.asList(blockFile));
         
         store = createStore(MAINNET, 10);
         resetStore(store);
-        chain = new FullPrunedBlockChain(context, store);
+        chain = new FullPrunedBlockChain(MAINNET, store);
```
Comment (msgilligan) [inline core/src/test/java/org/bitcoinj/core/AbstractFullPrunedBlockChainTest.java:242]: This comment isn't related to this PR, but I wanted to note there is potential confusion in unit tests between some of these `NetworkParameters` constants and `BitcoinNetwork` `enum` instances. As long as we avoid static imports of the `BitcoinNetwork` instances we should be fine.

File: core/src/main/java/org/bitcoinj/core/BlockChain.java [hunk 2]
```diff
@@ -51,42 +51,27 @@ public class BlockChain extends AbstractBlockChain {
      * {@link MemoryBlockStore} if you want to hold all headers in RAM and don't care about
      * disk serialization (this is rare).</p>
      */
-    public BlockChain(Context context, Wallet wallet, BlockStore blockStore) throws BlockStoreException {
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/BlockChain.java:54]: We might want to consider deprecating these constructors, but personally I'm fine with just removing them. Good riddance!

**Review by msgilligan — APPROVED** [review]
I approve. (Though I must confess I don't understand all the corner cases this might affect.)

**Comment by schildbach** [comment]
Merged.

**Pull request merged**